### PR TITLE
fix integer to enum conversion

### PR DESCRIPTION
**Pull request by maciejbocianski**
Fixes integer to enum conversion

### CODE REVIEW

**Comment by maciejbocianski** [comment]
@paul-szczepanek-arm

**Comment by paul-szczepanek-arm** [comment]
what was the error?

**Comment by maciejbocianski** [comment]
@paul-szczepanek-arm the problem is that it assigns to enum value from outside of enum range in this line:
https://github.com/ARMmbed/mbed-nfc-m24sr/blob/9ced51b3ae66e5533cbad0591813264bdedebb1b/m24sr_driver.cpp#L186

what then cause wrong comparison result in line:
https://github.com/ARMmbed/mbed-nfc-m24sr/blob/9ced51b3ae66e5533cbad0591813264bdedebb1b/m24sr_driver.cpp#L200

**Review by paul-szczepanek-arm — APPROVED** [review]

File: m24sr_driver.cpp [hunk 1]
```diff
@@ -193,15 +193,15 @@ static M24srError_t is_correct_crc_residue(uint8_t *data, uint8_t length) {
...
         }
     }
 
     if (status == NFC_COMMAND_SUCCESS) {
-        status = M24SR_SUCCESS;
+        return M24SR_SUCCESS;
     }
 
-    return status;
+    return M24SR_ERROR;
```
Comment (paul-szczepanek-arm) [inline m24sr_driver.cpp]: this now makes all errors look the same

Comment (maciejbocianski) [reply m24sr_driver.cpp]: @paul-szczepanek-arm 
changed to `return static_cast<M24srError_t>(status);`

**Review by paul-szczepanek-arm — APPROVED** [review]
add backing type for enum and the constant and it's good to go

**Comment by maciejbocianski** [comment]
@paul-szczepanek-arm updated

**Review by paul-szczepanek-arm — APPROVED** [review]

**Comment by paul-szczepanek-arm** [comment]
If there's a problem with an illegal error code it has to be checked higher up by whoever is calling it but this is still an improvement - merging.

**Pull request merged**